packages/realm-web/src/tests/User.test.ts

```diff
@@ -89,6 +90,37 @@ describe("User", () => {
                 },
             },
         ]);
+        // Expect the user to forget tokens anyway
+        expect(user.accessToken).equals(null);
+        expect(user.refreshToken).equals(null);
+    });
+
+    it("will forget tokens if session delete fails", async () => {
+        const app = new MockApp("my-mocked-app", [
+            LOCATION_RESPONSE,
+            new MongoDBRealmError(
+                "POST",
+                "http://localhost:1337/some-path",
+                401,
+                "",
+                "invalid session",
+            ),
+        ]);
+        const user = new User({
+            app,
+            id: "some-user-id",
+            accessToken: "deadbeef",
+            refreshToken: "very-refreshing",
+        });
+        // Log out the user
+        let errorThrown;
+        await user.logOut().then(null, err => {
+            errorThrown = err;
+        });
+        expect(errorThrown).instanceOf(MongoDBRealmError);
```

Comment:
Instead of mixing `await` and `then` (`catch`) chaining (which is a bit weird to read, for me at least), I would suggest:

```suggestion
        try {
            await user.logOut();
            assert.fail("Logout should fail");
        } catch (err) {
            expect(err).instanceOf(MongoDBRealmError);
        }
```